pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -827,6 +840,41 @@ func (r *ReconcileBareMetalHost) saveStatus(host *metal3v1alpha1.BareMetalHost)
 	return r.client.Status().Update(context.TODO(), host)
 }
 
+// Add time stamp at which soft power off command was sent to this host.
+func (r *ReconcileBareMetalHost) markLastSoftPowerOff(host *metal3v1alpha1.BareMetalHost) error {
+
+	if host.Status.LastSoftPowerOff == nil {
+		t := metav1.Now()
+		host.Status.LastSoftPowerOff = &t
+	}
+	return r.client.Status().Update(context.TODO(), host)
+}
+
+// Remove soft power off time stamp
+func (r *ReconcileBareMetalHost) unMarkLastSoftPowerOff(host *metal3v1alpha1.BareMetalHost) error {
+	host.Status.LastSoftPowerOff = nil
+	return r.client.Status().Update(context.TODO(), host)
+}
+
+// Check if Soft Power Off takes longer than timeout treshold, for any reason.
+func (r *ReconcileBareMetalHost) softPowerOffTimeoutNotReached(host *metal3v1alpha1.BareMetalHost) bool {
```

Comment:
Won't ironic manage the timeout for us?

Reply:
It does. But, we need to cover the case where ironic does not move to hard power off, for any reason. 
Basically, trying to enable the control to make the final say(regardless of ironic)